src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -0,0 +1,372 @@
+} from '.';
+
+export interface Props {
+  /**
+   * Handler for the "save" footer button
+   */
+  onSave: (indexPatternSpec: IndexPatternSpec) => void;
+  /**
+   * Handler for the "cancel" footer button
+   */
+  onCancel: () => void;
+  existingIndexPatterns: string[];
+  defaultTypeIsRollup?: boolean;
+  requireTimestampField?: boolean;
+}
+export interface TimestampOption {
+  display: string;
+  fieldName?: string;
+}
+
+export interface FormInternal extends Omit<IndexPatternConfig, 'timestampField'> {
+  timestampField?: TimestampOption;
+}
+
+const IndexPatternEditorFlyoutContentComponent = ({
+  onSave,
+  onCancel,
+  defaultTypeIsRollup,
+  requireTimestampField = false,
+}: Props) => {
+  const {
+    services: { http, indexPatternService, uiSettings },
+  } = useKibana<IndexPatternEditorContext>();
+
+  const { form } = useForm<IndexPatternConfig, FormInternal>({
+    defaultValue: {
+      type: defaultTypeIsRollup ? INDEX_PATTERN_TYPE.ROLLUP : INDEX_PATTERN_TYPE.DEFAULT,
+    },
+    schema,
+    onSubmit: async (formData, isValid) => {
+      if (!isValid) {
+        return;
+      }
+
+      const indexPatternStub: IndexPatternSpec = {
+        title: formData.title,
+        timeFieldName: formData.timestampField?.value,
+        id: formData.id,
+      };
+
+      if (type === INDEX_PATTERN_TYPE.ROLLUP && rollupIndex) {
+        indexPatternStub.type = INDEX_PATTERN_TYPE.ROLLUP;
+        indexPatternStub.typeMeta = {
+          params: {
+            rollup_index: rollupIndex,
+          },
+          aggs: rollupIndicesCapabilities[rollupIndex].aggs,
+        };
+      }
+
+      await onSave(indexPatternStub);
+    },
+  });
+
+  const { getFields } = form;
+
+  const [{ title, allowHidden, type }] = useFormData<FormInternal>({ form });
+  const [isLoadingSources, setIsLoadingSources] = useState<boolean>(true);
+
+  const [timestampFieldOptions, setTimestampFieldOptions] = useState<TimestampOption[]>([]);
+  const [isLoadingTimestampFields, setIsLoadingTimestampFields] = useState<boolean>(false);
+  const [isLoadingMatchedIndices, setIsLoadingMatchedIndices] = useState<boolean>(false);
+  const [allSources, setAllSources] = useState<MatchedItem[]>([]);
+  const [isLoadingIndexPatterns, setIsLoadingIndexPatterns] = useState<boolean>(true);
+  const [existingIndexPatterns, setExistingIndexPatterns] = useState<string[]>([]);
+  const [rollupIndex, setRollupIndex] = useState<string | undefined>();
+  const [
+    rollupIndicesCapabilities,
+    setRollupIndicesCapabilities,
+  ] = useState<RollupIndicesCapsResponse>({});
+  const [matchedIndices, setMatchedIndices] = useState<MatchedIndicesSet>({
+    allIndices: [],
+    exactMatchedIndices: [],
+    partialMatchedIndices: [],
+    visibleIndices: [],
+  });
+
+  // load all data sources and set initial matchedIndices
+  const loadSources = useCallback(() => {
+    getIndices(http, () => false, '*', allowHidden).then((dataSources) => {
+      setAllSources(dataSources);
```

Comment:
After mulling it over for a bit, I think you're right. It would be better for us to cancel our async requests but I don't think thats well supported here. I had to remind myself why exactly this isMounted check is here (maybe you're already familiar) - https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Reply:
addressed